backend/src/main/java/com/github/devtoju/backend/gaptext/GapTextContainerService.java

```diff
@@ -11,14 +12,25 @@
 @RequiredArgsConstructor
 public class GapTextContainerService {
     private final GapTextContainerRepo gapTextContainerRepo;
-    private final GapTextContainerMapper gapTextContainerMapper;
+    private final CreateDtoToGapTextContainerMapper createDtoToGapTextContainerMapper;
+    private final UpdateDtoToGapTextContainerMapper updateDtoToGapTextContainerMapper;
 
     public List<GapTextContainer> getAllContainers() {
         return gapTextContainerRepo.findAll();
     }
 
-    public GapTextContainer addContainer(GapTextContainerDTO newContainerDTO) {
-        var newContainer = gapTextContainerMapper.apply(newContainerDTO);
+    public GapTextContainer addContainer(GapTextContainerCreateDTO newCreateDTO) {
+        var newContainer = createDtoToGapTextContainerMapper.apply(newCreateDTO);
         return gapTextContainerRepo.save(newContainer);
     }
+
+    public GapTextContainer updateContainer(GapTextContainerUpdateDTO updateDTO) {
+        var containerNotExist = !gapTextContainerRepo.existsById(updateDTO.id());
```

Comment:
wusste nicht, dass es ein var keyword in java gibt, ist da auch nicht boolean besser?

Reply:
Da hatten wir doch ganz am Anfang mal drüber gesprochen. Das ist Ansichtssache. Ich mag es und daher verwende ich es in meinem Projekt auch. Zumindest dort, wo ich es gut verwenden kann. Es kommt auf den Kontext an und mit der if wird es klar:
```
 var containerNotExist = !gapTextContainerRepo.existsById(updateDTO.id());
  if (containerNotExist) {...
```
Bei Clean Code erstellt man gern eine Variable, die leichter lesbar ist und aussagt was es ist anstatt eine Bedingung direkt im if zu verwenden. Wenn ich den Code lese ist mir die Variable (fast) egal. Ich lese nur "if container not exist do something".
Falls es dich interessiert, hier ein netter Einstieg: https://www.youtube.com/watch?v=7EmboKQH8lM
Ich sag nur "like well written prose".